ambry-server/src/main/java/com/github/ambry/server/AmbryRequests.java

```diff
@@ -508,6 +519,122 @@ public void handleDeleteRequest(NetworkRequest request) throws IOException, Inte
+                .lifeVersion(MessageInfo.LIFE_VERSION_FROM_FRONTEND)
+                .build());
+          }
+          Store storeToDelete = storeManager.getStore(batchDeletePartitionRequestInfo.getPartition());
+          StoreBatchDeleteInfo storeBatchDeleteInfo = storeToDelete.batchDelete(infoList);
+          List<BlobDeleteStatus> blobDeleteStatuses = new ArrayList<>();
+          for (MessageErrorInfo messageErrorInfo : storeBatchDeleteInfo.getMessageErrorInfos()){
+            ServerErrorCode msgInfoServerErrorCode = messageErrorInfo.getError() == null ? ServerErrorCode.No_Error: ErrorMapping.getStoreErrorMapping(messageErrorInfo.getError());
+            blobDeleteStatuses.add(new BlobDeleteStatus((BlobId)messageErrorInfo.getMessageInfo().getStoreKey(), msgInfoServerErrorCode));
+          }
+          partitionResponseInfo = new BatchDeletePartitionResponseInfo(batchDeletePartitionRequestInfo.getPartition(), blobDeleteStatuses);
+
+          if (notification != null) {
+            for (StoreKey convertedStoreKey : convertedStoreKeys) {
+              notification.onBlobReplicaDeleted(currentNode.getHostname(), currentNode.getPort(),
+                  convertedStoreKey.getID(), BlobReplicaSourceType.PRIMARY);
+            }
+          }
+        }
+      } catch (Exception e) {
+        List<BlobDeleteStatus> blobDeleteStatuses = new ArrayList<>();
+        for (StoreKey storeKey : batchDeletePartitionRequestInfo.getBlobIds()){
+          blobDeleteStatuses.add(new BlobDeleteStatus((BlobId)storeKey, ServerErrorCode.Bad_Request));
+        }
+        // All blobs failed to delete
+        partitionResponseInfo = new BatchDeletePartitionResponseInfo(batchDeletePartitionRequestInfo.getPartition(), blobDeleteStatuses);
+        metrics.unExpectedStoreDeleteError.inc();
+      } finally {
+        long processingTime = SystemTime.getInstance().milliseconds() - startTime;
+        totalTimeSpent += processingTime;
+        publicAccessLogger.info("{} {} processingTime {}", batchDeleteRequest, response, processingTime);
+        // Update request metrics.
+        RequestMetricsUpdater metricsUpdater = new RequestMetricsUpdater(requestQueueTime, processingTime, 0, 0, false);
+        batchDeleteRequest.accept(metricsUpdater);
+        for (BlobDeleteStatus blobDeleteStatus: partitionResponseInfo.getBlobsDeleteStatus()){
+          // Set operation level error to Bad_Request if any of the blobs fails
+          ServerErrorCode error = blobDeleteStatus.getStatus();
+          if (error != ServerErrorCode.No_Error) {
+            // Handling for control test when request type is disabled + setting default to Bad_Request for any error.
+            operationServerErrorCode = error==ServerErrorCode.Temporarily_Disabled?ServerErrorCode.Temporarily_Disabled:ServerErrorCode.Bad_Request;
+            boolean logInErrorLevel = false;
+            if (error == ServerErrorCode.Blob_Not_Found) {
+              metrics.idNotFoundError.inc();
```

Comment:
Can we also please add one more kind of metrics that has counters at BatchDelete level? Will be useful to understand how the batchdelete is trending.